### PR TITLE
TEC-4037 Fix the block_categories function - 5.8 broke it.

### PR DESCRIPTION
**Pull request by Camwyn**
[TEC-4037]

Related:
https://github.com/the-events-calendar/events-pro/pull/1801
https://github.com/the-events-calendar/event-tickets/pull/2041

[TEC-4037]: https://theeventscalendar.atlassian.net/browse/TEC-4037

### CODE REVIEW

File: src/Tribe/Editor.php [hunk 1]
```diff
@@ -38,7 +38,14 @@ public function hook() {
 		add_action( 'admin_init', [ $this, 'assets' ] );
 
 		// Add Block Categories to Editor
-		add_action( 'block_categories', [ $this, 'block_categories' ], 10, 2 );
+		global $wp_version;
+		if ( ! class_exists( 'WP_Block_Editor_Context' ) || version_compare( $wp_version, '5.8', '<' ) ) {
+			// WP < 5.8
+			add_action( 'block_categories', [ $this, 'block_categories' ], 10, 2 );
+		} else {
+			// WP >= 5.8
```
Comment (lucatume) [inline src/Tribe/Editor.php:46]:
```suggestion
			// WP version is 5.8 or above.
```

File: src/Tribe/Editor.php [hunk 2]
```diff
@@ -561,6 +578,40 @@ public function block_categories( $categories, $post ) {
 		);
 	}
 
+	/**
+	 * Add "Event Blocks" category to the editor
```
Comment (lucatume) [inline src/Tribe/Editor.php:582]:
```suggestion
	 * Add "Event Blocks" category to the editor.
```

File: src/Tribe/Editor.php [hunk 3]
```diff
@@ -38,7 +38,14 @@ public function hook() {
 		add_action( 'admin_init', [ $this, 'assets' ] );
 
 		// Add Block Categories to Editor
-		add_action( 'block_categories', [ $this, 'block_categories' ], 10, 2 );
+		global $wp_version;
+		if ( ! class_exists( 'WP_Block_Editor_Context' ) || version_compare( $wp_version, '5.8', '<' ) ) {
```
Comment (lucatume) [inline src/Tribe/Editor.php:42]: The `version_compare` check is cheaper and we're using the `class_exists`, here, just to run the same check on the WordPress version. Drop the `class_exist` and just run the `version_compare`?

File: src/Tribe/Editor.php [hunk 4]
```diff
@@ -543,9 +550,19 @@ public function deregister_scripts() {
...
+	 * @param array<array<string|string>> $categories An array of categories each an array
+	 *                                                in the format property => value.
+	 * @param WP_Post                     $post       The post object we're editing.
+	 *
 	 * @return array
 	 */
 	public function block_categories( $categories, $post ) {
+		// Handle where someone is using this outside of this object
+		global $wp_version;
+		if ( class_exists( 'WP_Block_Editor_Context' ) || version_compare( $wp_version, '5.8', '>=' ) ) {
```
Comment (lucatume) [inline src/Tribe/Editor.php:562]: As above: should we drop the `class_exists` check here and just run the `version_compare`?

**Review by lucatume — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor.php (hunk 4 above)
Comment (lucatume) [reply src/Tribe/Editor.php:562]: I would drop the `class_exists` check completely, and use `$context instanceof WP_Block_Editor_Context` right in the method that needs it.

File: src/Tribe/Editor.php (hunk 3 above)
Comment (Camwyn) [reply src/Tribe/Editor.php:42]: Yeah, the class_exists check is the "official" recommended one - but I found it unreliable. I can drop it.

File: readme.txt [hunk 5]
```diff
@@ -223,7 +223,7 @@ Remember to always make a backup of your database and files before updating!
 
 = [5.9.0] TBD =
 
-
+* Fix - Add a new function to add "Event Blocks" category to the editor for WP >= 5.8. [TBD]
```
Comment (borkweb) [inline readme.txt:226]:
```suggestion
* Fix - Add a new function to add "Event Blocks" category to the editor for WP >= 5.8. [TEC-4037]
```

**Review by borkweb — APPROVED** [review]
:+1: once tests pass!

**Pull request merged**